### DISCUSSION TITLE
Cors Error

### DISCUSSION THREAD

**Discussion by deldonk1123** [vercel/next.js#39477] [Help] [answered]
Hi, I wrote this code in next config js file.
```typescript
...
headers() {
    return [
      {
        source: "/api/:path*",
        headers: [
          { key: "Access-Control-Allow-Credentials", value: "true" },
          { key: "Access-Control-Allow-Origin", value: "*" },
          {
            key: "Access-Control-Allow-Methods",
            value: "GET,OPTIONS,PATCH,DELETE,POST,PUT",
          },
          {
            key: "Access-Control-Allow-Headers",
            value:
              "X-CSRF-Token, X-Requested-With, Accept, Accept-Version, Content-Length, Content-MD5, Content-Type, Date, X-Api-Version",
          },
        ],
      },
    ];
  },
...
```
but I get cors error.
![image](https://user-images.githubusercontent.com/93312414/184023346-cba812d9-4ae1-42a4-bf5c-deb53d56e9b9.png)

I am wrong? how can I fix it? Hope your help. Thanks in advance!!

**Comment by gaston-flores** [comment, marked as answer]
Hey @klausrichard420! It looks like you are making a request to Firebase directly from your page, without going through Next's API routes. I think you need to configure CORS in your Firebase bucket, not in Next.

**Reply by deldonk1123** [reply]
Hi, Thanks for your reply.
Do you know how to configure CORS in firebase bucket?

**Reply by gaston-flores** [reply]
I think you need to [follow these steps.](https://firebase.google.com/docs/storage/web/download-files#cors_configuration)

**Reply by deldonk1123** [reply]
Thanks @gaston-flores 
It works perfect. good work.
